src/event_handlers.py

```diff
@@ -0,0 +1,140 @@
+# pylint: disable=too-few-public-methods
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Union
+
+from celery import Task
+from loguru import logger
+
+from src.constants import EventEnum, EventType, LabelName
+from src.helpers import get_exception_class
+from src.instrumentation import EventCounter, EventGauge
+
+
+class InvalidTaskEventLabelName(Exception):
+    pass
+
+
+class IEventHandler(ABC):
```

Comment:
Base abstract class defining interface for the handlers. All common logic should go to this in the future.